File: make_array_empty/MakeArrayEmpty.java
class MakeArrayEmpty {
    public long countOperationsToEmptyArray(int[] nums) {
        if (nums == null || nums.length == 0) {
            return 0;
        }

        TreeMap<Integer, Integer> table = new TreeMap<>();
        for (int i = 0; i < nums.length; i++) {
            table.put(nums[i], i);
        }

        Map.Entry<Integer, Integer> first = table.firstEntry();
        int lastPosition = first.getValue();
        long steps = nums.length;
        table.remove(first.getKey());
        int idx = 1;
        for (Map.Entry<Integer, Integer> entry : table.entrySet()) {
            if (entry.getValue() < lastPosition) {
                steps += nums.length - idx;
            }
            lastPosition = entry.getValue();
            idx++;
        }

        return steps;
    }
}

File: make_array_empty/Make_Array_Empty.cpp
class Make_Array_Empty {
public:
    long long countOperationsToEmptyArray(vector<int>& nums) {
        if (nums.empty()) {
            return 0;
        }

        unordered_map<int, int> table;
        for (int i = 0; i < nums.size(); i++) {
            table[nums[i]] = i;
        }
        sort(nums.begin(), nums.end());

        long long steps = nums.size();
        for (int i = 1; i < nums.size(); i++) {
            if (table[nums[i]] >= table[nums[i - 1]]) {
                continue;
            }
            steps += nums.size() - i;
        }
        return steps;
    }
};